autorest/azure/token.go

```diff
@@ -301,14 +301,17 @@ func (spt *ServicePrincipalToken) refreshInternal(resource string) error {
 	}
 
 	var newToken Token
+	var aadError AADServiceError
 	err = autorest.Respond(resp,
 		autorest.WithErrorUnlessOK(),
-		autorest.ByUnmarshallingJSON(&newToken),
+		autorest.ByUnmarshallingSuccessandErrorJSON(&newToken, &aadError),
```

Comment:
@jkapil I **strongly** believe this needs to be changed to `WithAADServiceErrorUnlessOK` as I suggested. That is much more in keeping with the go-autorest model.